icom/icom.c

```diff
@@ -3289,15 +3289,33 @@ int icom_set_powerstat(RIG *rig, powerstat_t status)
 	}
 
 	// we can ignore this retval
-	// sending more than enough 0xfe's to take up the rs232
+	// sending more than enough 0xfe's to wake up the rs232
 	icom_transaction(rig, 0xfe, 0xfe, fe_buf, fe_len,
 					ackbuf, &ack_len);
 	retval = icom_transaction(rig, C_SET_PWR, pwr_sc, NULL, 0,
 					ackbuf, &ack_len);
+	rig_debug(RIG_DEBUG_VERBOSE, "%s #2 called retval=%d\n", __func__,retval);
+  int i=0;
+  int retry = 3/rig->state.rigport.retry;
+  if (status==RIG_POWER_ON) { // wait for wakeup only
+    for(i=0;i<retry;++i) { // up to 10 attempts
```

Comment:
I think that this kind of retry mechanism should be implemented on a higher level -- not in rig/manufacturer-specific code. This commit already implements it by copy-pasting code for two manufacturers. Also, while it may be a special case, all rigs may not support reading frequency.